### ISSUE TITLE
mdconvert doesn't work with the latest version of NetCDF4

### ISSUE THREAD

**Issue by xiki-tempula** [mdtraj/mdtraj#1939]
when running `mdconvert T4lys_ligand_solvated_pruned.nc -o ddddsdss.xtc` and using NetCDF4 as backend, mdconvert would give the error of

```
Traceback (most recent call last):
  File "/Users/zwu/miniconda3/envs/mdtraj/bin/mdconvert", line 11, in <module>
    sys.exit(entry_point())
             ^^^^^^^^^^^^^
  File "/Users/zwu/miniconda3/envs/mdtraj/lib/python3.11/site-packages/mdtraj/scripts/mdconvert.py", line 597, in entry_point
    main(args)
  File "/Users/zwu/miniconda3/envs/mdtraj/lib/python3.11/site-packages/mdtraj/scripts/mdconvert.py", line 324, in main
    with InFileFormat(fn, "r") as infile:
         ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/zwu/miniconda3/envs/mdtraj/lib/python3.11/site-packages/mdtraj/formats/netcdf.py", line 169, in __init__
    self._handle = netcdf(filename, mode=mode, **input_args)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: netcdf_file.__init__() got an unexpected keyword argument 'format'
```
The environment is created with `conda create -n mdtraj -c conda-forge mdtraj NetCDF4`

conda list

```
astunparse                1.6.3              pyhd8ed1ab_2    conda-forge
blosc                     1.21.6               h5499902_0    conda-forge
bzip2                     1.0.8                h99b78c6_7    conda-forge
c-ares                    1.34.2               h7ab814d_0    conda-forge
c-blosc2                  2.15.1               h5063078_0    conda-forge
ca-certificates           2024.8.30            hf0a4a13_0    conda-forge
certifi                   2024.8.30          pyhd8ed1ab_0    conda-forge
cftime                    1.6.4           py311h0f07fe1_1    conda-forge
hdf4                      4.2.15               h2ee6834_7    conda-forge
hdf5                      1.14.4          nompi_h99fbd1e_101    conda-forge
icu                       75.1                 hfee45f7_0    conda-forge
krb5                      1.21.3               h237132a_0    conda-forge
libaec                    1.1.3                hebf3989_0    conda-forge
libblas                   3.9.0           25_osxarm64_openblas    conda-forge
libcblas                  3.9.0           25_osxarm64_openblas    conda-forge
libcurl                   8.10.1               h13a7ad3_0    conda-forge
libcxx                    19.1.2               ha82da77_0    conda-forge
libedit                   3.1.20191231         hc8eb9b7_2    conda-forge
libev                     4.33                 h93a5062_2    conda-forge
libexpat                  2.6.3                hf9b8971_0    conda-forge
libffi                    3.4.2                h3422bc3_5    conda-forge
libgfortran               5.0.0           13_2_0_hd922786_3    conda-forge
libgfortran5              13.2.0               hf226fd6_3    conda-forge
libiconv                  1.17                 h0d3ecfb_2    conda-forge
libjpeg-turbo             3.0.0                hb547adb_1    conda-forge
liblapack                 3.9.0           25_osxarm64_openblas    conda-forge
libnetcdf                 4.9.2           nompi_h853a48d_115    conda-forge
libnghttp2                1.64.0               h6d7220d_0    conda-forge
libopenblas               0.3.28          openmp_h517c56d_0    conda-forge
libsqlite                 3.47.0               hbaaea75_0    conda-forge
libssh2                   1.11.0               h7a5bd25_0    conda-forge
libxml2                   2.12.7               h01dff8b_4    conda-forge
libzip                    1.11.1               hfc4440f_0    conda-forge
libzlib                   1.3.1                h8359307_2    conda-forge
llvm-openmp               19.1.2               hb52a8e5_0    conda-forge
lz4-c                     1.9.4                hb7217d7_0    conda-forge
mdtraj                    1.10.0          py311habad988_0    conda-forge
ncurses                   6.5                  h7bae524_1    conda-forge
netcdf4                   1.7.2           nompi_py311hd13e3db_100    conda-forge
numexpr                   2.10.1          py311hd9fdc8a_3    conda-forge
numpy                     1.26.4          py311h7125741_0    conda-forge
openssl                   3.3.2                h8359307_0    conda-forge
packaging                 24.1               pyhd8ed1ab_0    conda-forge
pandas                    2.2.3           py311h9cb3ce9_1    conda-forge
pip                       24.2               pyh8b19718_1    conda-forge
py-cpuinfo                9.0.0              pyhd8ed1ab_0    conda-forge
pyparsing                 3.2.0              pyhd8ed1ab_1    conda-forge
pytables                  3.10.1          py311hc9a4827_3    conda-forge
python                    3.11.10         hc51fdd5_3_cpython    conda-forge
python-dateutil           2.9.0              pyhd8ed1ab_0    conda-forge
python-tzdata             2024.2             pyhd8ed1ab_0    conda-forge
python_abi                3.11                    5_cp311    conda-forge
pytz                      2024.1             pyhd8ed1ab_0    conda-forge
readline                  8.2                  h92ec313_1    conda-forge
scipy                     1.14.1          py311hf1db568_1    conda-forge
setuptools                75.1.0             pyhd8ed1ab_0    conda-forge
six                       1.16.0             pyh6c4a22f_0    conda-forge
snappy                    1.2.1                hd02b534_0    conda-forge
tk                        8.6.13               h5083fa2_1    conda-forge
typing-extensions         4.12.2               hd8ed1ab_0    conda-forge
typing_extensions         4.12.2             pyha770c72_0    conda-forge
tzdata                    2024b                hc8b5060_0    conda-forge
wheel                     0.44.0             pyhd8ed1ab_0    conda-forge
xz                        5.2.6                h57fd34a_0    conda-forge
zlib                      1.3.1                h8359307_2    conda-forge
zlib-ng                   2.2.2                hf9b8971_0    conda-forge
zstd                      1.5.6                hb46c0d2_0    conda-forge
```

**Comment by mattwthompson** [comment]
1.7.2 was released last night, so perhaps this is a genuine and new breakage? I don't see anything crazy in the changelog: https://github.com/Unidata/netcdf4-python/blob/d203d8bf1455fb85df67abae73556216af6cc597/Changelog#L1-L8

https://prefix.dev/channels/conda-forge/packages/netcdf4

**Comment by xiki-tempula** [comment]
No 1.7.1 is also broken if you do 
mdconvert [mdcrd.nc](https://github.com/mdtraj/mdtraj/blob/master/tests/data/mdcrd.nc) -o eee.xtc

**Comment by mattwthompson** [comment]
Ah, I see, I didn't look so closely in the traceback and that you're only using `mdconvert`

**Comment by xiki-tempula** [comment]
Ok, it seems that this problem is fixed in the 1.10.1 release. Do you mind release the new patch release? Thanks!
